crates/sequencer/src/compat.rs

```diff
@@ -189,25 +189,25 @@ where
 
     pub(crate) async fn produce_l2_block_pre_fork2(
         &mut self,
-        da_block: Da::FilteredBlock,
+        current_l1_block: Da::FilteredBlock,
         l1_fee_rate: u128,
         l2_block_mode: &L2BlockMode,
         last_used_l1_height: &mut u64,
     ) -> anyhow::Result<(u64, StateDiff)> {
         let active_fork_spec = self.fork_manager.active_fork().spec_id;
 
-        let da_height = da_block.header().height();
-        let (l2_height, l1_height) = match self
+        let current_l1_height = current_l1_block.header().height();
+        let (soft_confirmation_l2_height, soft_confirmation_l1_height) = match self
             .ledger_db
             .get_head_soft_confirmation()
             .map_err(|e| anyhow!("Failed to get head soft confirmation: {}", e))?
         {
             Some((l2_height, sb)) => (l2_height.0 + 1, sb.da_slot_height),
-            None => (1, da_height),
+            None => (1, current_l1_height),
         };
         anyhow::ensure!(
-            l1_height == da_height || l1_height + 1 == da_height,
-            "Sequencer: L1 height mismatch, expected {da_height} (or {da_height}-1), got {l1_height}",
+            soft_confirmation_l1_height == current_l1_height || soft_confirmation_l1_height + 1 == current_l1_height,
+            "Sequencer: L1 height mismatch, expected {current_l1_height} (or {current_l1_height}-1), got {soft_confirmation_l1_height}",
```

Comment:
```suggestion
            "Sequencer: L1 height mismatch, expected {current_l1_height} (or {current_l1_height - 1}), got {soft_confirmation_l1_height}",
```
What about this insteaad ?

Reply:
I don't want to execute the subtraction.. similar to how the code was originally.